internal/pulsartest/app.go

```diff
@@ -36,19 +37,39 @@ func New(params Params) (*App, error) {
 		return nil, err
 	}
 
-	producerName := fmt.Sprintf("pulsartest-%s", serverId)
-	producer, err := pulsarClient.CreateProducer(pulsar.ProducerOptions{
-		Name:             producerName,
-		CompressionType:  compressionType,
-		CompressionLevel: compressionLevel,
-		Topic:            params.Pulsar.JobsetEventsTopic,
-	})
-	if err != nil {
-		return nil, errors.Wrapf(err, "error creating pulsar producer %s", producerName)
+	var producer pulsar.Producer
+	var reader pulsar.Reader
+
+	if cmdType == "submit" {
```

Comment:
Testme?

Reply:
Tests to exercise the parameters-checking logic in `New(...)` have been added in `e2e/pulsartest_client/app_test.go` at commit 27472bc4 - the test is there because the Pulsar client library needs a running Pulsar server when instantiating a `Producer` object (it checks if the topic is valid, etc). This test is run as part of the `tests-e2e` makefile target.